actions/requestActions.js

```diff
@@ -0,0 +1,71 @@
+import fetch from 'isomorphic-fetch';
+import { API_HOST } from '../actions';
+
+const url = '/api/request_job';
+
+const START = 'yasp/request/START';
+const ERROR = 'yasp/request/ERROR';
+const PROGRESS = 'yasp/request/PROGRESS';
+const MATCH_ID = 'yasp/request/MATCH_ID';
+
+export const requestActions = {
+  START,
+  ERROR,
+  PROGRESS,
+  MATCH_ID,
+};
+
+const setMatchId = (matchId) => ({
+  type: MATCH_ID,
+  matchId,
+});
+
+const requestStart = () => ({
+  type: START,
+});
+
+const requestError = (error) => ({
+  type: ERROR,
+  error,
+});
+
+const requestProgress = (progress) => ({
+  type: PROGRESS,
+  progress,
+});
+
+const requestSubmit = (matchId) => (dispatch) => {
+  const formData = new FormData();
+  formData.append('match_id', matchId);
+  dispatch(requestStart());
+  return fetch(`${API_HOST}${url}`, {
+    method: 'post',
+    body: formData,
+  })
+  .then(res => res.json())
+  .then((json) => {
+    function poll() {
```

Comment:
don't define this function inside of your `then()`. Just declare it above the method and then call it with dispatch as an argument. This way we aren't defining this function every time you dispatch this action, and it's also easier to read.